utils/src/formatting_tools.c

```diff
@@ -316,3 +319,95 @@ int concat(uint8_t ** dest, size_t * dest_length, uint8_t * input,
   *dest_length = new_dest_len;
   return (0);
 }
+
+//############################################################################
+// convert_string_to_digest()
+//############################################################################
+int convert_string_to_digest(char *str, TPM2B_DIGEST * digest)
+{
+  // substring to holds 2 hex values at a time
+  char substr[3];
+  substr[2] = '\0';
+
+  size_t strlength = strlen(str);
+
+  if (strlength != (size_t) (2 * KMYTH_DIGEST_SIZE) )
+  {
+    kmyth_log(LOG_ERR, "invalid input string length ... exiting");
+    return 1;
+  }
+
+  if (digest == NULL || digest->buffer == NULL )
+  {
+    kmyth_log(LOG_ERR, "invalid digest argument ... exiting");
+    return 1;
+  }
+
+  // initializes buffer with all 0 hex values
```

Comment:
I think the code here is fine, but the comment no longer matches (i.e., expectedPolicyBuffer is allocated, but not initialized to all-zero hex values)